### ISSUE TITLE
No Debug Target Available 

### ISSUE THREAD

**Issue by MoseySauce** [microsoft/vscode-dotnettools#411]
### Describe the Issue

I have an intermittent issue with debugging a .Net MAUI project using VSCODE on Linux (Ubuntu 23.04).

what happens:  Pressing "Run and Debug" or F5 to start a debug session will initiate the build, and the build is successful.  It will launch the Android emulator (even if it's not currently running).  After some time the debug session will end and in the Debug Console it will state "No debug target available, skipping debugging."  This appears to be an intermittent problem because I first encountered this last week, but the other day I was able to debug without issue, but again tonight I can no longer debug the Maui project.  I have tried restarting my laptop, cleaning solution, deleting .vscode, obj, bin folders, and deleting the Android emulator and recreating a new one.  I note that the .Net Maui extension (Preview v0.3.12) is asking me to update my Android SDK, but I installed the SDK via Android Studio and I do not know what else to do about that.  I also note that my emulator is listed in the bottom right corner of VSCODE, and I suspect this is why the emulator lanuches successfully.  It just seems that the debugger cannot hook into the emulator process in some manner.

### Steps To Reproduce

I was following a simple tutorial to make a simple .Net Maui app.  I was able to debug without issue initially.  The issue is intermittent and tonight I am completly unable to debug with a running Android emulator.

### Expected Behavior

I expect to be able to start a debug session with a Android Emulator and successfully hit a debug breakpoint.

### Environment Information

Ubuntu 23.04 Lunar Lobster

VSCODE:
Version: 1.81.1
Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
Date: 2023-08-09T22:18:39.991Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Linux x64 6.0.0-1020-oem snap

.Net Maui: 0.3.12

> VS bug [#1874873](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1874873)

**Comment by MoseySauce** [comment]
I wanted to note that earlier today I set up my home computer (Intel Nuc) to also develop .Net Maui. It is Debian 12, and all works as expected there.  I noticed that on my laptop (Ubuntu 23.04) I was tying to use a Pixel 7 Pro API 33 emulator, where my Debian machine has a Pixel 5 API 33 emulator.  As a test I created a new Pixel 5 API 33 emulator on my laptop and can now successfully build and debug the project.  

As a test, I switched my debug target back to the Pixel 7 Pro, and was then unable to debug with the same symptoms described before.  I then switched back to the Pixel 5 and was again unable to debug as described.  I closed the emulator and vscode then restarted vscode.  After it examined the maui project directory, I ensured my debug target was the Pixel 5 and I *was* able to debug again.  The error appears to be dependent on the emulator the debugger is targeting, which I wasn't expecting because both are targeting API 33.

**Comment by oceanthrsty** [comment]
I have also been having this same intermittent issue.  Yesterday it was running fine, we have been making updates.  But not sure what we did that changed it.
I am running .Net7 Maui on VS Code for Mac.

I get "build successful" but then in the debug console I get "no debug target available, skipping debug".

But, I am able to debug with Visual Studio Mac, and of course in VS windows.

This is what is being run by VScode:
`dotnet build -t:Build -p:Configuration=Debug -f net7.0-maccatalyst /Users/mattw/proj/GFITpro.csproj`

**Comment by gkarabin** [comment]
I ran into the same with VS Code and .NET 8 rc2. I had been using .NET 8 rc1 fine for some time. I installed rc2 yesterday. It worked into the evening, but this morning after restarting VS Code this began happening.

**Comment by WolfBublitz** [comment]
I have the same problem with a maui blazor project running with .NET 8 on Mac OS Sonoma (arm).

**Comment by WolfBublitz** [comment]
Same project on Windows is working!

**Comment by WolfBublitz** [comment]
Are there any news regarding this issue? I have tried the latest version of the C# and C# DevKit Extensions but does not work :-(. This really annoying and makes developing blazor maui impossible on macos.

**Comment by BlueJay2022** [comment]
Same problem, all new setup on new Mac Pro m2
VS Code, maui workload, .net 8

**Comment by leang97** [comment]
Same problem, MAUI blazor hybrid app on a Macbook Air M2 chip.
VS Code, New MAUI was created into a .net standard project with Winform.
Any workarounds and solution currently available?

**Comment by niguiecandy** [comment]
I'm on a M1 Macbook Pro.
The VS Code shows the startup devices as this:
<img width="602" alt="image" src="https://github.com/microsoft/vscode-dotnettools/assets/39695857/2fc23e63-ade2-4847-862f-5a78b387b063">
It says 'Build only', does that mean I can't debug my app? why can't I select the machine that I'm running the build as a target?

**Comment by fernandez271** [comment]
This guide worked for me
https://zhuanlan.zhihu.com/p/667482446

Basically 
- cmd + shift + p
- .NET MAUI: Pick MacOS Device
- Select My Mac

**Comment by MoseySauce** [comment]
I've still encountered this bug off and on these past months.  I don't think there is a single root cause, because what works one time sometimes doesn't work the next.

**Comment by tondat** [comment]
We've made some improvements around the debug targets detection in the latest prerelease version of the .NET MAUI Dev Kit extension (v 0.11.34). Could you give it a try and let us know if you are still experiencing this issue? And please do attach the full content of the `.NET MAUI` output window.

**Comment by mauroa** [comment]
We are unable to continue the investigation without the required information and also, as mentioned above, there have been many fixes around the debug targets and the debug experience, that might fix your issues. 
Feel free to request a reopening if you still experience any issue after updating to latest C#, C# DevKit and .NET MAUI versions and also please provide the information requested above.
Thanks for your help and feedback.

**Comment by thomas51516** [comment]
Hellooo!,

I’m facing the same issue on Linux Ubuntu.
I created an emulator and also connected my phone.
When I run the adb devices command, both devices are displayed. However, when I click on "Debug" in VS Code, neither of the devices is shown. In the end, I always get the same message:
No debug target available, skipping debugging.

I have installed all the recommended extensions:

.NET MAUI
C# DevKit
C# IntelliCode
Here are the versions I’m using:

ADB Version: Version 35.0.2-12147458
Android Debug Bridge: Version 1.0.41
.NET Version: 9.0.102

![Image](https://github.com/user-attachments/assets/d96b9cd1-0c4a-4483-9ec0-9efb04eb03c0)
